packages/client/src/__tests__/getPrismaClient.dataproxy.test.ts

```diff
@@ -0,0 +1,137 @@
+import { getConfig, getDMMF, parseEnvValue } from '@prisma/sdk'
+import crypto from 'crypto'
+
+import { chinook } from '../fixtures/chinook'
+import type { GetPrismaClientConfig } from '../runtime/getPrismaClient'
+import { getPrismaClient } from '../runtime/getPrismaClient'
+
+const schemaContentsBase64 = Buffer.from(chinook).toString('base64')
+const schemaContentsHashed = crypto.createHash('sha256').update(schemaContentsBase64).digest('hex')
+let prismaClientOptions: GetPrismaClientConfig
+let config
+
+beforeAll(async () => {
+  config = await getConfig({ datamodel: chinook })
+  const prismaClientDmmf = await getDMMF({ datamodel: chinook })
+
+  prismaClientOptions = {
+    inlineSchema: schemaContentsBase64,
+    inlineSchemaHash: schemaContentsHashed,
+    document: prismaClientDmmf,
+    generator: {
+      name: 'client',
+      provider: {
+        value: 'prisma-client-js',
+        fromEnvVar: null,
+      },
+      config: { engineType: 'dataproxy' },
+      output: null,
+      binaryTargets: [],
+      previewFeatures: [],
+    },
+    // clientVersion: '',
+    dirname: './some/path',
```

Comment:
Seems this one is actually relevant some way. https://github.com/prisma/prisma/pull/12738#issuecomment-1108594156

Reply:
Yes, the override of the ENGINE_TYPE via the env vars in CI caused problems here